packages/design-system/src/components/Dialog/Dialog.tsx

```diff
@@ -123,6 +123,11 @@ export const Dialog = (props: DialogProps) => {
 
   useBodyScrollPrevention(modalProps.isOpen ?? true);
 
+  // 🥑
+  // Is `tabIndex` necessary here? It's not clear why it's being set to -1.
+  // It's also not clear why the `role` attribute is being set to `document`.
+  // Should this allow for more flexible heading tags? Locked into `h1` for now.
+  // Can we remove `role="main"` from the `main` element?
   return (
     <NativeDialog
```

Comment:
TODO:
1. Replace `tabIndex` with `autofocus` attr
2. Replace `<h1>` with `<h2>`
3. Remove `role="main"`

I don't believe the Bootstrap issue exists on native dialog elements, so I don't believe the `role="document"` is needed, but removing this role would have a cascade effect on the other attrs on this `div` (`aria-labelledby` would need to be addressed).